src/main/java/io/xdb/core/client/Client.java

```diff
@@ -46,10 +56,55 @@ public String startProcess(
         return startProcessInternal(processType, processId, input);
     }
 
+    /**
+     * Stop a process execution as TERMINATED within the default namespace.
+     *
+     * @param processId a unique identifier used to differentiate between different executions of the same process type.
+     */
+    public void stopProcess(final String processId) {
+        stopProcess(DEFAULT_NAMESPACE, processId, ProcessExecutionStopType.TERMINATE);
+    }
+
+    /**
+     * Stop a process execution within the default namespace.
+     *
+     * @param processId a unique identifier used to differentiate between different executions of the same process type.
+     * @param stopType  specify how the process execution should be stopped, either as TERMINATED or FAILED.
+     *
+     */
+    public void stopProcess(final String processId, final ProcessExecutionStopType stopType) {
+        stopProcess(DEFAULT_NAMESPACE, processId, stopType);
+    }
+
+    /**
+     * Stop a process execution.
+     *
+     * @param namespace the namespace in which the operation should be performed.
+     * @param processId a unique identifier used to differentiate between different executions of the same process type.
+     * @param stopType  specify how the process execution should be stopped, either as TERMINATED or FAILED.
+     *
+     */
+    public void stopProcess(final String namespace, final String processId, final ProcessExecutionStopType stopType) {
```

Comment:
Actually we should let users set namespace in client options like cadence temporal. Otherwise it’s too cumbersome to do it everywhere

Reply:
True. I will make another MR to add it into client options.